File: 367. Valid Perfect Square.py
class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        # 1 + 3 + 5
        if num == 0:
            return True

        cnt, ele = 0, 1
        while cnt < num:
            cnt, ele = cnt + ele, ele + 2
            # print cnt, ele
        return cnt == num

File: 367. Valid Perfect Square.java
public class Solution {
    public boolean isPerfectSquare(int num) {
        if (num < 0) return false;
        if (num == 0 || num == 1) return true;
        long low = 0, hi = num;
        while (low <= hi) {
            long mid = low + (hi - low) / 2;
            if (mid * mid == num) return true;
            if (mid * mid < num) {
                low = mid + 1;
            } else {
                hi = mid - 1;
            }
        }
        return false;
    }

    
}